src/capi/azext_capi/templates/AzureMachinePool.yaml

```diff
@@ -12,4 +12,6 @@ spec:
       diskSizeGB: 30
       managedDisk:
         storageAccountType: Premium_LRS
+    {%- if AZURE_SSH_PUBLIC_KEY_B64 %}
```

Comment:
Do we need to omit this if it's empty? I think in the CAPZ templates we do `${AZURE_SSH_PUBLIC_KEY_B64:=""}`.